src/ring_theory/ideal/operations.lean

```diff
@@ -1604,3 +1611,86 @@ begin
 end
 
 end ring_hom
+
+namespace double_quot
+open ideal
+variables {R : Type u} [comm_ring R] (I J : ideal R)
+
+/-- define `quot_left_to_quot_sum` to be the obvious ring hom `R/I → R/(I ⊔ J)` -/
+def quot_left_to_quot_sum : I.quotient →+* (I ⊔ J).quotient :=
+  ideal.quotient.inclusion I (I ⊔ J) (le_sup_left)
+
+/-- This will be used to lift `quot_left_to_quot_sum` to a map `(R/I)/J' → R/(I ⊔ J)`-/
+lemma img_left_in_ker (x : I.quotient) (hx : x ∈ J.map(ideal.quotient.mk I)) :
+  quot_left_to_quot_sum I J x = 0 :=
+begin
+  have hIJmap: ((quot_left_to_quot_sum I J).comp(ideal.quotient.mk I) '' J) =
+    (ideal.quotient.mk (I ⊔ J) '' J),
+   {apply set.ext,
+    intro y,
+    split,
+
+     {intro hy,
+      obtain ⟨z, hz⟩ := (set.mem_image ((quot_left_to_quot_sum I J).comp(ideal.quotient.mk I))
+        J y).1 hy,
+      unfold quot_left_to_quot_sum at hz,
+      rw [ring_hom.comp_apply, ideal.quotient.inclusion, ideal.quotient.lift_mk] at hz,
+      rw ← hz.right,
+      exact set.mem_image_of_mem (ideal.quotient.mk (I ⊔ J)) hz.left},
+
+     {intro hy,
+      obtain ⟨z, hz⟩ := (set.mem_image (ideal.quotient.mk (I ⊔ J)) J y).1 hy,
+      rw [quot_left_to_quot_sum, set.mem_image_eq],
+      use z,
+      rwa [ring_hom.comp_apply, ideal.quotient.inclusion, ideal.quotient.lift_mk]},
+    },
+
+  have hJ: (J.map (ideal.quotient.mk I)).map (quot_left_to_quot_sum I J) = J.map
+    (ideal.quotient.mk (I ⊔ J))
+    := by rw [ideal.map_map, ideal.map,hIJmap, ← ideal.map],
+
+  have hmapx : quot_left_to_quot_sum I J x ∈ (J.map (ideal.quotient.mk I)).map
+    (quot_left_to_quot_sum I J),
+    {rw ideal.map,
+    apply set.mem_of_subset_of_mem (ideal.subset_span),
+    exact (set.mem_image_of_mem (quot_left_to_quot_sum I J) hx)},
+
+  have : J.map(ideal.quotient.mk(I ⊔ J)) = ⊥ :=  map_mk_eq_bot_of_le le_sup_right,
+  rwa [hJ,this] at hmapx,
+end
+
+/-- define `double_quot_to_quot_add` to be the induced ring hom `(R/I)/J' ->R/(I ⊔ J)`,
+  where `J'` is the image of `J` in `R/I` -/
+def double_quot_to_quot_add : (J.map (ideal.quotient.mk I)).quotient →+* (I ⊔ J).quotient :=
+ideal.quotient.lift (ideal.map (ideal.quotient.mk I) J) (quot_left_to_quot_sum I J)
+ (img_left_in_ker I J)
+
+/-- define `double_quot_mk` to be the composite of the maps `R → (R/I) and (R/I) → (R/I)/J'` -/
+def double_quot_mk : R →+* (J.map I^.quotient.mk).quotient:=
+((J.map I^.quotient.mk)^.quotient.mk).comp I^.quotient.mk
+
+-- Another short result for lifting map `ring_to_double_quot` to a map `R/(I ⊔ J) → (R/I)/J'`
+lemma mem_add_double_quot_mk (x : R) (hx : x ∈ I ⊔ J) : double_quot_mk I J x = 0 :=
+begin
+  have hIJtoJ : (I ⊔ J).map(ideal.quotient.mk I) = J.map(ideal.quotient.mk I) := by {
```

Comment:
```suggestion
lemma sup_le_ker_double_quot_mk : I ⊔ J ≤ (double_quot_mk I J).ker :=
begin
  intros x hx,
  have hIJtoJ : (I ⊔ J).map(ideal.quotient.mk I) = J.map(ideal.quotient.mk I) := by {
```
You'll have to add a `ring_hom.mem_ker` to the `rw` below.

Reply:
Better yet, with that statement you can make a much shorter proof:
```
begin
  apply sup_le _ _,
  { intros x hx,
    apply quotient.eq_zero_iff_mem.mpr,
    rw quotient.eq_zero_iff_mem.mpr hx,
    exact ideal.zero_mem _, },
  { intros x hx,
    apply quotient.eq_zero_iff_mem.mpr,
    exact ideal.mem_map_of_mem _ hx, },
end
```

Reply:
In fact, you can prove the stronger statement that `I ⊔ J = (double_quot_mk I J).ker`, but I'll leave that to you as an exercise because my proof is awful.